nixos/modules/services/audio/mpdscribble.nix

```diff
@@ -0,0 +1,183 @@
+{ config, lib, pkgs, ... }:
+
+with lib; with types;
+
+let
+  cfg = config.services.mpdscribble;
+  mpdCfg = config.services.mpd;
+
+  endpointUrls = {
+    "last.fm" = "http://post.audioscrobbler.com";
+    "libre.fm" = "http://turtle.libre.fm";
+    "jamendo" = "http://postaudioscrobbler.jamendo.com";
+    "listenbrainz" = "http://proxy.listenbrainz.org";
+  };
+
+  mkSection = secname: cfg:
+    ''
+      [${secname}]
+      url      = ${cfg.url}
+      username = ${cfg.username}
+      password = ${cfg.password}
+      journal  = ${cfg.journal}
+    '';
+
+  endpoints = concatStringsSep "\n"
+    (
+      mapAttrsToList
+        mkSection
+        cfg.endpoints
+    );
+
+  cfgFile = pkgs.writeText "mpdscribble.conf" ''
+    ## mpdscribble - an audioscrobbler for the Music Player Daemon.
+    ## http://mpd.wikia.com/wiki/Client:mpdscribble
+
+    # HTTP proxy URL.
+    ${optionalString (!isNull cfg.proxy) "proxy = ${cfg.proxy}"}
+
+    # The location of the mpdscribble log file.  The special value
+    # "syslog" makes mpdscribble use the local syslog daemon.  On most
+    # systems, log messages will appear in /var/log/daemon.log then.
+    # "-" means log to stderr (the current terminal).
+    log = ${cfg.logFile}
+
+    # How verbose mpdscribble's logging should be.  Default is 1.
+    verbose = ${toString cfg.verbose}
+
+    # How often should mpdscribble save the journal file? [seconds]
+    journal_interval = ${toString cfg.journalInterval}
+
+    # The host running MPD, possibly protected by a password
+    # ([PASSWORD@]HOSTNAME).
+    host = ${cfg.host}
```

Comment:
@Sohalt I was trying to suggest, that mpd's password shouldn't be plain visible here, and more over, at the store. You should use a `environment.etc.` file, with the proper permissions for the `mpdscribble` user. But, this is a dumb request when we do write to the store the mpd passwords - without https://github.com/NixOS/nixpkgs/pull/95599 . Could you give me a hand there with a review / test please?